jpos/src/main/java/org/jpos/q2/Q2.java

```diff
@@ -771,10 +785,7 @@ private void parseCmdLine (String[] args) {
                 pidFile = line.getOptionValue("p");
             if (line.hasOption("n"))
                 name = line.getOptionValue("n");
-            if (line.hasOption("E")) {
-                System.setProperty("jpos.env", line.getOptionValue("E"));
-                Environment.getEnvironment();
```

Comment:
So you removed the required initial call to `Environment.getEnvironment()` ?

Reply:
This call was being made only for option  `E`
I moved this code to the top, plus the one that parses the new option `Ed` (for the envdir)
Then I added
```java
if (line.hasOption("Ed") || line.hasOption("E")) {
    Environment.reload();
}
```

The `reload()` method is returning the `INSTANCE` as the `getEnvironment()` was doing, but only after reloading with the new settings.  I put that at the beginning, before the other cmd line options are parsed, to give it the opportunity to have the env loaded (and from the proper file!), before other parts of the system

If neither `E` nor `Ed` is explicited, whoever gets the environment first in the global code would be initializing the static `INSTANCE` anyway (from the default file).  That's still the same that was happening before.

I hope my logic is clear, and especially hope it's correct :-)